host/profiles.go

```diff
@@ -0,0 +1,190 @@
+	allowedDevices := append(*job.Config.AllowedDevices, []*configs.Device{
+		{
+			Path:        "/dev/kvm",
+			Type:        'c',
+			Major:       kvmDev.major,
+			Minor:       kvmDev.minor,
+			Permissions: "rwm",
+		},
+		{
+			Path:        "/dev/net/tun",
+			Type:        'c',
+			Major:       tunDev.major,
+			Minor:       tunDev.minor,
+			Permissions: "rwm",
+		},
+	}...)
+	job.Config.AllowedDevices = &allowedDevices
+
+	// auto create /dev/kvm and /dev/net/tun
+	autoCreatedDevices := append(*job.Config.AutoCreatedDevices, []*configs.Device{
+		{
+			Path:        "/dev/kvm",
+			Type:        'c',
+			Major:       kvmDev.major,
+			Minor:       kvmDev.minor,
+			Permissions: "rwm",
+		},
+		{
+			Path:        "/dev/net/tun",
+			Type:        'c',
+			Major:       tunDev.major,
+			Minor:       tunDev.minor,
+			Permissions: "rwm",
+		},
+	}...)
+	job.Config.AutoCreatedDevices = &autoCreatedDevices
+
+	// allow the job to create a network TAP interface
+	linuxCapabilities := append(*job.Config.LinuxCapabilities, "CAP_NET_ADMIN")
+	job.Config.LinuxCapabilities = &linuxCapabilities
+
+	return nil
+}
+
+const LOOP_CTL_GET_FREE = 0x4C82
+
+func jobProfileLoop(job *host.Job) error {
+	// find an available loop device using /dev/loop-control
+	f, err := os.OpenFile("/dev/loop-control", os.O_RDONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	index, err := unix.IoctlGetInt(int(f.Fd()), LOOP_CTL_GET_FREE)
+	if err != nil {
+		return err
+	}
+
+	// load the device
+	loopDev, err := loadDevice(fmt.Sprintf("/sys/class/block/loop%d/dev", index))
+	if err != nil {
+		return fmt.Errorf("error loading loop device: %s", err)
+	}
+
+	// allow the loop device as /dev/loop0
+	allowedDevices := append(*job.Config.AllowedDevices, []*configs.Device{
+		{
+			Path:        "/dev/loop0",
+			Type:        'b',
+			Major:       loopDev.major,
+			Minor:       loopDev.minor,
+			Permissions: "rwm",
+		},
+	}...)
+	job.Config.AllowedDevices = &allowedDevices
+
+	// auto create /dev/loop0
+	autoCreatedDevices := append(*job.Config.AutoCreatedDevices, &configs.Device{
+		Path:        "/dev/loop0",
+		Type:        'b',
+		Major:       loopDev.major,
+		Minor:       loopDev.minor,
+		Permissions: "rwm",
+	})
+	job.Config.AutoCreatedDevices = &autoCreatedDevices
+
+	return nil
+}
+
+type device struct {
+	major int64
+	minor int64
+}
+
+func loadDevice(path string) (*device, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	s := strings.SplitN(strings.TrimSpace(string(data)), ":", 2)
+	if len(s) != 2 {
+		return nil, fmt.Errorf("unexpected data in %s: %q", path, data)
+	}
+	dev := &device{}
+	dev.major, err = strconv.ParseInt(s[0], 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing device major number from %q: %s", data, err)
+	}
+	dev.minor, err = strconv.ParseInt(s[1], 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing device minor number from %q: %s", data, err)
```

Comment:
These two errors should include the path that failed to parse as well.

Reply:
Done in b0a97b9.